velox/exec/Operator.cpp

```diff
@@ -158,9 +158,16 @@ Operator::translators() {
 std::unique_ptr<Operator> Operator::fromPlanNode(
     DriverCtx* ctx,
     int32_t id,
-    const core::PlanNodePtr& planNode) {
+    const core::PlanNodePtr& planNode,
+    std::shared_ptr<ExchangeClient> exchangeClient) {
   for (auto& translator : translators()) {
-    auto op = translator->toOperator(ctx, id, planNode);
+    std::unique_ptr<Operator> op;
+    if (planNode->requiresExchangeClient()) {
+      op = translator->toOperator(ctx, id, planNode, exchangeClient);
```

Comment:
```
std::move(exchangeClient)
```

Reply:
We cannot move here. Check my prev explanation:
> Yeah that's right we only want exchange client to be in one operator. But here it is iterating through every registered translators to find out the right one (that returns non-null ptr). If we move here then the first iteration will destroy the exchangeClient pointer and the following iterations won't have access to exchangeClient anymore.